### ISSUE TITLE
Problem with a conditional test in rarecurve()

### ISSUE THREAD

**Issue by gavinsimpson** [vegandevs/vegan#112]
This showed up whilst checking the current github master branch of **vegan**. I don't have time to look at this just now, but we're seeing a lot of warnings like this

```r
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
```

indicating that `sample` or `minsample` (probably the former given the object names) is of length > 1. The full Example check log output for this example is shown below

```r
> ### Name: diversity
> ### Title: Ecological Diversity Indices and Rarefaction Species Richness
> ### Aliases: diversity rarefy rrarefy drarefy rarecurve fisher.alpha
> ###   specnumber
> ### Keywords: univar
> 
> ### ** Examples
> 
> data(BCI)
> H <- diversity(BCI)
> simp <- diversity(BCI, "simpson")
> invsimp <- diversity(BCI, "inv")
> ## Unbiased Simpson of Hurlbert 1971 (eq. 5):
> unbias.simp <- rarefy(BCI, 2) - 1
> ## Fisher alpha
> alpha <- fisher.alpha(BCI)
> ## Plot all
> pairs(cbind(H, simp, invsimp, unbias.simp, alpha), pch="+", col="blue")
> ## Species richness (S) and Pielou's evenness (J):
> S <- specnumber(BCI) ## rowSums(BCI > 0) does the same...
> J <- H/log(S)
> ## beta diversity defined as gamma/alpha - 1:
> data(dune)
> data(dune.env)
> alpha <- with(dune.env, tapply(specnumber(dune), Management, mean))
> gamma <- with(dune.env, specnumber(dune, Management))
> gamma/alpha - 1
       BF        HF        NM        SF 
0.5483871 0.6666667 1.6250000 1.2909091 
> ## Rarefaction
> (raremax <- min(rowSums(BCI)))
[1] 340
> Srare <- rarefy(BCI, raremax)
> plot(S, Srare, xlab = "Observed No. of Species", ylab = "Rarefied No. of Species")
> abline(0, 1)
> rarecurve(BCI, step = 20, sample = raremax, col = "blue", cex = 0.6)
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
Warning in if (sample > minsample) warning(gettextf("sample = %d is larger than smallest site maximum (%d)",  :
  the condition has length > 1 and only the first element will be used
```

**Comment by jarioksa** [comment]
This was caused by commit 53587b2a95. Got to fix it in `rarefy` (the change is justified, but need be executed more robustly).

**Comment by jarioksa** [comment]
The **bug** of conditional test was fixed in a40d1f5e0c. The **feature** of warning on requesting subsample larger than the original community still remains. This may trigger warnings in `rarecurve`. It seems that the old behaviour was documented (and hence intended): The `man/diversity.Rd` says:
```
The size of
 \code{sample} should be smaller than total community size, but the
 function will silently work for larger \code{sample} as well and
 return non-rarefied species richness (and standard error = 0).
```
I haven't changed documentation yet,  because we may cancel the new policy of warning. But let's see.

**Comment by jarioksa** [comment]
The 991ab91 fix seems to cure `example(rarecurve)` problems. Are there any cases where adding warning in `rarefy`  causes problems, or should we accept the change and update `man/diversity.Rd`?
